tests/unit/test_tools.py

```
@@ -134,96 +135,442 @@ def test_calculator_handles_basic_expression(self):
+        mock_ddgs_instance.text.return_value = iter([
+            {'title': 'Result', 'href': 'https://example.com', 'body': 'Description'}
+        ])
+
+        result = search_web("test query")
 
+        assert 'Result' in result
+        assert mock_ddgs.call_count == 2
+
+    @patch('time.sleep')
+    @patch('ddgs.DDGS')
+    def test_search_web_non_ssl_error_reraised(self, mock_ddgs, mock_sleep):
```

Ruff (0.14.6)

Unused method argument: `mock_sleep`

(ARG002)